lib/compress/inference.js

```diff
@@ -238,13 +237,29 @@ export function is_nullish(node, compressor) {
             && (fixed = node.definition().fixed) instanceof AST_Node
             && is_nullish(fixed, compressor)
         )
-        // Recurse into those optional chains!
-        || node instanceof AST_PropAccess && node.optional && is_nullish(node.expression, compressor)
-        || node instanceof AST_Call && node.optional && is_nullish(node.expression, compressor)
-        || node instanceof AST_Chain && is_nullish(node.expression, compressor)
     );
 }
 
+// Find out if this expression is chained from a base-point that we can
+// statically analyze as null or undefined.
+function is_nullish_shortcircuit(node, compressor) {
+    if (node instanceof AST_PropAccess || node instanceof AST_Call) {
+        return (
+            (node.optional && is_null_or_undefined(node.expression, compressor))
+            || is_nullish_shortcircuit(node.expression, compressor)
+        );
+    }
+    if (node instanceof AST_Chain) return is_nullish_shortcircuit(node.expression);
```

Comment:
```suggestion
    if (node instanceof AST_Chain) return is_nullish_shortcircuit(node.expression, compressor);
```

Missing an argument here.